Add request triggers docs to Java Profiler (#113854)
Azure/azure-content: articles/azure-monitor/app/java-standalone-profiler.md

The Application Insights Java profiler uses the JFR profiler provided by the JVM to record profiling data, allowing users to download the JFR recordings at a later time and analyze them to identify the cause of performance issues.

> This data is gathered on demand when trigger conditions are met. The available triggers are thresholds over CPU usage and Memory consumption, and Request (SLA triggers).

Reviewer: I would probably be a bit more specific as many may not know what is meant by "SLA triggers". So might split it up into an extra sentence, i.e:
```
This data is gathered on demand when trigger conditions are met. The available triggers are thresholds over CPU usage and Memory consumption, and Request (SLA triggers). Request triggers monitor Spans generated by Open Telemetry and allow the user to configure SLA requirements over the duration of those Spans.
```

> This data is gathered on demand when trigger conditions are met. The available triggers are thresholds over CPU usage and Memory consumption, and Request (SLA triggers). Request triggers monitor Spans generated by Open Telemetry and allow the user to configure SLA requirements over the duration of those Spans.

Reviewer 2: ```suggestion
This data is gathered on demand when trigger conditions are met. The available triggers are thresholds over CPU usage, Memory consumption, and Request (SLA triggers). Request triggers monitor Spans generated by Open Telemetry and allow the user to configure SLA requirements over the duration of those Spans.
```

For more detailed description of the various triggers available, see [profiler overview](../profiler/profiler-overview.md).

> The ApplicationInsights Java Agent monitors CPU, memory and request duration (e.g., a business transaction) and if it breaches a configured threshold a profile is triggered. 

Reviewer 2: ```suggestion
The ApplicationInsights Java Agent monitors CPU, memory, and request duration such as a business transaction. If it breaches a configured threshold, a profile is triggered. 
```

> The ApplicationInsights Java Agent monitors CPU and memory consumption and request duration (e.g., a business transaction) and if it breaches a configured threshold a profile is triggered. 

Reviewer: comma for "CPU and memory"

- Full garbage collection is executed
- The Tenured regions occupancy is above 691 mb after collection

### Request

SLA triggers are based on OpenTelemetry (otel) and they will initiate a profile if certain criteria is fulfilled.

Each individual trigger configuration is formed as follows:

- `Name` - A unique identifier for the trigger.
- `Filter` - Filters the requests of interest for the trigger.
- `Aggregation` - This calculates the ratio of requests that breached a given threshold.
  - `Threshold` - A value (in milliseconds) above which a request breach is determined.
  - `Minimum samples` - The minimum number of samples that must be collected for the aggregation to produce data, this is to prevent triggering off of small sample sizes.
  - `Window` - Rolling time window (in milliseconds).
- `Threshold` - The threshold value (percentage) applied to the aggregation output. If this value is exceeded, a profile is initiated.

> For instance, consider a scenario where you want to trigger a profile when more than 30% of requests to a specific endpoint (/users/.*) take longer than 50 ms within a 10000 ms window.

Reviewer: Were you intending to add an example below this?

Author: I added it

       
    2. Select "Triggers"
    

>     3. Configure the required CPU and Memory thresholds, or add request triggers (if enabled) and select Apply.

Reviewer: `CPU, Memory or request triggers(if enabled)`

- `profile`. Uses the `profile.jfc` jfc configuration that ships with JFR.
- A path to a custom jfc configuration file on the file system, i.e `/tmp/myconfig.jfc`.

`enableRequestTriggering` This will let know the profiler if JFR profiling should be triggered based on request configuration. 
This value can be one of:

> - `true` Profiling will be triggered if a request trigger criteria are met.

Reviewer: change `criteria are met` to `thresholds are breached`

- `profile`. Uses the `profile.jfc` jfc configuration that ships with JFR.
- A path to a custom jfc configuration file on the file system, i.e `/tmp/myconfig.jfc`.

> `enableRequestTriggering` This will let know the profiler if JFR profiling should be triggered based on request configuration. 

Reviewer 3: Suggest changing this to: "Whether JFR profiling should be triggered based on request configuration."
